refactor #44 : UserService Refactoring

sproutboot opened the pull request:
- validation 의존성 추가
- 커스텀 예외 처리
- 매직 넘버에서 상수로 전환(공백, 길이)

sproutboot committed:
refactor #44 : UserService Refactoring
- validation 의존성 추가
- 커스텀 예외 처리
- 매직 넘버에서 상수로 전환(공백, 길이)

f-lab-joseph commented on src/main/java/com/yummy/delivery/service/UserService.java:
```diff
@@ -49,35 +54,35 @@ public void userWithdrawal(@Param("email") String email){
     public void checkPasswordLength(UserDTO userDTO){
-        if(userDTO.getPassword().length() < 8){
-            throw new IllegalStateException("비밀번호를 8자리 이상 입력해주세요.");
+        if(userDTO.getPassword().length() < EIGHT_LENGTH_CHECK){
```
변수명이 적절치 않은 것 같습니다. 어떤 역할을 하는 상수인지를 더 잘 나타낼 방법이 있을 것 같아요. 
그리고 지금 대부분의 DTO 값 검증은 Spring validator 로 대체가 가능한 부분일 것 같습니다!

sproutboot committed:
refactor #44 : UserService Refactoring
- @RestControllerAdvice를 이용해 예외처리
- 컨트롤러에 @Valid 어노테이션 추가
- Spring validator로 DTO값 검증
- 기존 UserDTO 클래스에서 UserRequestDTO로 클래스 이름 변경

sproutboot committed:
Merge branch 'main' into feature/44-userservice-modify
